README.md

```diff
@@ -248,3 +248,38 @@ doxygen docs/doxygen/config.doxyfile
 
 See [CONTRIBUTING.md](./.github/CONTRIBUTING.md) for information on
 contributing.
+
+## Using Setter and Getter Functions for Persistent Storage
+
+The coreMQTT library provides setter and getter functions to allow the application to store and restore MQTT state in persistent memory. This is useful for handling QoS2 messages after a device reboot.
+
+### Setter Function
+
+The `MQTT_SetOutgoingPublishRecord` function allows the application to set an outgoing publish record in the MQTT context. It can be used to restore the state of the MQTT context after a device reboot.
```

Comment:
I wonder whether a callback might be more suitable here?

Here is what i am thinking: The method described here will work perfectly fine when the reboot is planned. But what if the reboot is unplanned such as a power failure or link to the internet goes down.

In such cases, application might not have enough time to ping the library repeatedly for new packets. I understand that there might be race condition even in callback. But it is less pronounced.

When the application calls the API described above, it can either do that very frequently leading to higher CPU consumption, or it can do that slowly leading to a higher chance of a missed QoS2 publish when a reboot happens.

Reply:
I thought more about it and I would like to fix my error in the above statement.

In case of unplanned power loss/reboot, the coreMQTT library cannot be held responsible for keeping the actual state and the stored state in NVM in sync. Because regardless of whatever the library does, there will always be a race condition in the actual and stored state of the system. 

Thus, I think the library should only support planned reboots. In case of unplanned reboots, I think the application knows/understands the system best and it should try to fix any issues with the state of the system.

I will think more on this. But let me know if you have any suggestions or concerns @vishwamartur.

Thanks